File: C#/Valid_Anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length) {
            return false;
        }
        
        int[] charCounts = new int[26];
        
        for (int i = 0; i < s.Length; i++) {
            charCounts[s[i] - 'a']++;
            charCounts[t[i] - 'a']--;
        }
        
        foreach (int count in charCounts) {
            if (count != 0) {
                return false;
            }
        }
        
        return true;
    }
}

File: Cpp/Valid_Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) {
            return false;
        }
        int count[26] = {0};
        for (char c : s) {
            count[c - 'a']++;
        }
        for (char c : t) {
            count[c - 'a']--;
            if (count[c - 'a'] < 0) {
                return false;
            }
        }
        return true;
    }
};